projects/go-lib/src/lib/components/go-select/go-select.component.ts

```diff
@@ -34,22 +47,137 @@ export class GoSelectComponent extends GoFormBaseComponent implements OnInit {
   @Input() virtualScroll: boolean = false;
 
   @Output() scrollToEnd: EventEmitter<any> = new EventEmitter<any>();
-  @Output() scroll: EventEmitter<{ start: number, end: number }> = new EventEmitter<{ start: number; end: number }>();
+  @Output() scroll: EventEmitter<{ start: number, end: number }> = new EventEmitter<{ start: number, end: number }>();
 
   @ContentChild('goSelectOption') goSelectOption: TemplateRef<any>;
   @ContentChild('goSelectOptionGroup') goSelectOptionGroup: TemplateRef<any>;
   @ContentChild('goSelectSelectedOption') goSelectSelectedOption: TemplateRef<any>;
 
+  private controlSubscription: Subscription;
+  // store refined items after search
+  refinedItems: any[] = [];
+  // stores previous selected items when typeahead is enabled only in case of selectAll.
+  private previousSelectedItems: any[] = [];
+
   ngOnInit(): void {
     this.closeOnSelect = this.multiple ? false : this.closeOnSelect;
+    this.handleControlInitialValue();
+    this.subscribeToControlChanges();
+  }
+
+  ngOnDestroy(): void {
+    this.controlSubscription.unsubscribe();
+  }
+
+  private subscribeToControlChanges(): void {
+    if (this.multiple && this.showSelectAll) {
+      this.controlSubscription = this.control.valueChanges.subscribe((value: any) => {
+        this.handleMultipleControlChanges(value);
+      });
+    }
+  }
+
+  private handleMultipleControlChanges(value: any): void {
+    this.emptyRefinedItems();
+    if (!value?.length) {
+      this.resetTypeaheadItems();
+    }
   }
 
   onSelectAll(): void {
-    this.control.patchValue(this.items.map((item: any) => this.bindValue ? item[this.bindValue] : item));
+    if (this.typeahead) {
```

Comment:
Hi @swimtech20 typeahead is an Input decorator. Changing variables name would be a breaking change. I would suggest we keep this variable as it is since ng-select accepts ```typeahead```.

Reply:
I see what you mean, now. This is out of scope for this bug fix, so let's not worry about it!